extensions/big-data-cluster/src/bigDataCluster/dialog/bdcDashboard.ts

```diff
@@ -65,19 +65,23 @@ export class BdcDashboard {
 
 			// Refresh button
 			const refreshButton = modelView.modelBuilder.button()
-				.withProperties({
+				.withProperties<azdata.ButtonProperties>({
 					label: localize('bdc.dashboard.refreshButton', "Refresh"),
-					iconPath: IconPathHelper.refresh,
-					height: '50px'
+					iconPath: IconPathHelper.refresh
 				}).component();
 
-			refreshButton.onDidClick(() => this.model.refresh());
+			refreshButton.onDidClick(async () => {
+				refreshButton.iconPath = IconPathHelper.refresh_rotate;
+				refreshButton.enabled = false;
+				await this.model.refresh();
```

Comment:
What if this errors? Do we need to try/catch?

Reply:
Also, could we put this onclick handling logic in a separate method? Would make it easier to test in the future when we focus on code coverage.